File: FSharp/Day02/Program.fs
open System
open AdventUtilities
open CodeChecker

let example = InputData().ReadAllText 2 "example1"
let input = InputData().ReadAllText 2 "input"

let part1 = invalidCode isInvalidSplit input
let part2 = invalidCode isInvalidChunked input

printfn "Part 1: %i" part1
printfn "Part 2: %i" part2

File: CSharp/Day02/Program.cs

using AdventUtilities;
using Day02;

Dictionary<string, string> filePaths = new()
{
    ["example"] = Path.Combine(InputData.GetSolutionDirectory(), "InputFiles", "Day02", "example1.txt"),
    ["input"] = Path.Combine(InputData.GetSolutionDirectory(), "InputFiles", "Day02", "input.txt")
};

string example = File.ReadAllText(filePaths["example"]);
string input = File.ReadAllText(filePaths["input"]);

Console.WriteLine($"Part 1: {CodeChecker.SumInvalidCodeSplits(input)}");
Console.WriteLine($"Part 2: {CodeChecker.SumInvalidCodes(input)}");

